### PR TITLE
Support LUKS unlock via SSH

### PR DESCRIPTION
**Pull request by rfan1**
SLE-16506 - support LUKS unlock via SSH (via dracut-ssh)

- Related ticket: https://progress.opensuse.org/issues/81780
- Needles: n/a
- Verification run:
https://openqa.suse.de/tests/8704498
https://openqa.suse.de/tests/8710449
https://openqa.suse.de/tests/8710450

I don't plan to automate it on other platforms since it is not hardware specific case, I will try to automate it on TW later as well.

### CODE REVIEW

File: data/autoyast_sle15/autoyast_btrfs_luks1_separate_boot.xml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+          <partition_nr t="integer">1</partition_nr>
+          <resize t="boolean">false</resize>
+          <size>209715200</size>
+        </partition>
+        <partition t="map">
+          <create t="boolean">true</create>
+          <create_subvolumes t="boolean">false</create_subvolumes>
+          <crypt_key>nots3cr3t</crypt_key>
+          <crypt_method t="symbol">luks1</crypt_method>
+          <filesystem t="symbol">btrfs</filesystem>
```
Comment (lilyeyes) [inline data/autoyast_sle15/autoyast_btrfs_luks1_separate_boot.xml:74]: In test case https://bugzilla.suse.com/tr_show_case.cgi?case_id=1768639
There is one NOTE: "Do not use BtrFS file system" (I really forgot why)
But from the verification run, it is OK.

Comment (rfan1) [reply data/autoyast_sle15/autoyast_btrfs_luks1_separate_boot.xml:74]: "/boot must not be a sub volume on an encrypted btrfs"

I think I meet the requirement above :)

**Comment by lilyeyes** [comment]
Do you want to cover negative test, such as step "10. try wrong password, the decryption should be failed "?

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Review by wangqianzou — APPROVED** [review]
LGTM

**Review by Amrysliu — COMMENTED** [review]
LGTM

**Review by Amrysliu — APPROVED** [review]

File: tests/security/luks1/unlock_via_ssh_server.pm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    mutex_create('SERVER_UP');
+
+    my $children = get_children();
+    mutex_wait('CLIENT_READY', (keys %$children)[0]);
+
+    # Modify the grub with static ip address
+    add_grub_cmdline_settings('ip=10.0.2.101::10.0.2.1:255.255.255.0:eth0::off rd.neednet=1', update_grub => 1);
+
+    # dracut
```
Comment (jouyingbin) [inline tests/security/luks1/unlock_via_ssh_server.pm:35]: tiny comment,
Would you add more description here, but it's not a big deal.

**Review by jouyingbin — APPROVED** [review]
LGTM.

**Pull request merged**